### PR TITLE
MAID-1996 avoid tunnelling when proxying

### PR DESCRIPTION
**Pull request by maqi**

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @afck

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: tests/mock_crust/tunnel.rs [hunk 1]
```diff
@@ -270,3 +270,35 @@ fn tunnel_node_blocked() {
...
+
+#[test]
+fn avoid_tunnelling_when_proxying() {
+    let min_section_size = 5;
+    let network = Network::new(min_section_size, None);
+    network.block_connection(Endpoint(2), Endpoint(3));
+    network.block_connection(Endpoint(3), Endpoint(2));
+    let mut nodes = create_connected_nodes(&network, min_section_size);
+    verify_invariant_for_all_nodes(&nodes);
+    assert!(unwrap!(locate_tunnel_node(&nodes, PeerId(2), PeerId(3))) != 0);
```
Comment (dhardy) [inline tests/mock_crust/tunnel.rs]: Why `!= 0`? Maybe I'm misreading this, but 0 is the index of the first node, which is what we'd normally use, right?

File: tests/mock_crust/tunnel.rs [hunk 2]
```diff
@@ -270,3 +270,35 @@ fn tunnel_node_blocked() {
...
+        .config(config.clone())
+        .endpoint(endpoint)
+        .cache(false)
+        .create());
+    network.block_connection(Endpoint(nodes.len() - 1), Endpoint(0));
+    network.block_connection(Endpoint(0), Endpoint(nodes.len() - 1));
+    poll_and_resend(&mut nodes, &mut []);
+    verify_invariant_for_all_nodes(&nodes);
+    assert_eq!(nodes.len() - 2,
+               unwrap!(locate_tunnel_node(&nodes, PeerId(0), PeerId(nodes.len() - 1))));
```
Comment (dhardy) [inline tests/mock_crust/tunnel.rs]: You could instead write `assert!(locate_tunnel_node(..) == Some(nodes.len() - 2))`

File: tests/mock_crust/tunnel.rs (hunk 1 above)
Comment (maqi) [inline tests/mock_crust/tunnel.rs]: we use node 0 as proxy, and with this change, proxying node shall not be used as tunnel_node any more, so here is to assert this.

File: tests/mock_crust/tunnel.rs (hunk 2 above)
Comment (maqi) [inline tests/mock_crust/tunnel.rs]: just prefer the value of two sides can be printed in case of failure

**Pull request merged**